javalib/src/main/scala/java/util/LinkedList.scala

```diff
@@ -348,17 +353,55 @@ class LinkedList[E]() extends AbstractSequentialList[E]
 
   def descendingIterator(): Iterator[E] = {
     new Iterator[E] {
-      private val iter = listIterator(_size)
+
+      private var last: Double = -1
+      private var i: Double = _size
+
+      private var currentNode: Node[E] = null
+
+      private var lastNode: Node[E] = LinkedList.this.last
```

Comment:
You could write this class using only the `currentNode`, there is no real need for the `lastNode`. If you initialize `private var currentNode: Node[E] = LinkedList.this.last` you can implement `next` as:
```scala
  ... // check hasNext, update i and last
  val nextVlaue = currentNode.vlaue
  currentNode = currentNode.prev
  nextVlaue
}
```
and `remove` with:
```scala
def remove(): Unit = {
  checkThatHasLast()
  val toBeRemoved = currentNode.next
  if (toBeRemoved.next eq null) removeLast()
  else removeNode(toBeRemoved)
  ... // update i and last
}
```